ui/src/app/views/workspace-share/component.spec.ts

```diff
@@ -147,5 +147,58 @@ describe('WorkspaceShareComponent', () => {
         .toEqual(WorkspaceAccessLevel.OWNER);
   }));
 
+  it('validates and allows usernames', fakeAsync(() => {
+    spyOn(workspaceSharePage.fixture.componentRef.instance.workspacesService, 'shareWorkspace')
+      .and.callThrough();
+
+    const userValues = {
+      workspaceEtag: undefined,
+      items: [
+        { email: 'sampleuser1@fake-research-aou.org', role: 'OWNER' },
+        { email: 'sampleuser2@fake-research-aou.org', role: 'WRITER' },
+        { email: 'sampleuser3@fake-research-aou.org', role: 'READER' },
+        { email: 'sampleuser4@fake-research-aou.org', role: 'WRITER' }
+      ]
+    };
 
+    workspaceSharePage.fixture.componentRef.instance.profileStorageService.reload();
+    workspaceSharePage.readPageData();
+    simulateInput(workspaceSharePage.fixture, workspaceSharePage.emailField, 'sampleuser4');
+    workspaceSharePage.fixture.componentRef.instance.setAccess('Writer');
+
+
+    simulateClick(workspaceSharePage.fixture,
+      queryByCss(workspaceSharePage.fixture, '.add-button'));
+    workspaceSharePage.readPageData();
+    expect(workspaceSharePage.fixture.componentRef.instance.workspacesService.shareWorkspace)
+        .toHaveBeenCalledWith('defaultNamespace', '1', userValues);
+  }));
+
+  it('validates and allows email addresses', fakeAsync(() => {
+    spyOn(workspaceSharePage.fixture.componentRef.instance.workspacesService, 'shareWorkspace')
+      .and.callThrough();
+
+    const userValues = {
+      workspaceEtag: undefined,
+      items: [
+        { email: 'sampleuser1@fake-research-aou.org', role: 'OWNER' },
+        { email: 'sampleuser2@fake-research-aou.org', role: 'WRITER' },
+        { email: 'sampleuser3@fake-research-aou.org', role: 'READER' },
+        { email: 'sampleuser4@fake-research-aou.org', role: 'WRITER' }
+      ]
+    };
+
+    workspaceSharePage.fixture.componentRef.instance.profileStorageService.reload();
```

Comment:
Is this needed? If so, can it be moved into the initial `beforeEach()` setup if it is needed? Conceptually it doesn't seem like it should be needed since the component should be able to render in isolation (perhaps something to change in profile service stub?)

Reply:
This is needed because the share workspace page uses a stale profile, so it needs to know to update without any info published yet. Moved to before each.